### DISCUSSION TITLE
How to observe panel state changes

### DISCUSSION THREAD

**Discussion by kaz080** [scenee/FloatingPanel#460] [Q&A] [answered]
Thank you for such a nice package!

Is there a way to observe panel state changes, such as moving from .half to .full?
I'd like to send an analytics event when users move a floating panel from half to full.

**Comment by scenee** [comment, marked as answer]
Yes, there is. You are able to use `floatingPanelDidChangePosition` delegate method like [this](https://github.com/scenee/FloatingPanel#add-tap-gestures-to-the-surface-view).

Note: `floatingPanelDidChangePosition` are going be `floatingPanelDidChangeState` since v2.4.0 or later.

**Reply by kaz080** [reply]
It works well!
Thank you so much!!
